### PR TITLE
refactor: improve placeholder CLI

### PR DESCRIPTION
**Pull request by namit-chandwani**
## Description
I reinitialized the CLI using Cobra as I found it to be much easier than making changes to the older one.

## Changes made

- Added `extensions` as a parent command. For example - the command: `yourCLi list` will now be: `yourCLi extensions list`.
- Changed the entire structure of CLI
- Some minor fixes

**PS: The LICENSE file was added by Cobra on using the `cobra init` command. Please let me know if I need to remove it.**

-----

In my next PR, I will be making some more changes to the CLI like:
- Adding flags to some commands for accepting input. For example: Accepting the name and version values through the command - `yourCLi extensions install <name>@<version>`
- Cleaning up the code by removing some unwanted init() functions and flags
- Giving proper descriptions to each command and even adding it to the documentation

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.charmil.yaml)")
+
+	// Cobra also supports local flags, which will only run
+	// when this action is called directly.
+	rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
```
Comment (wtrocki) [inline cmd/cmd.go:52]: What we need config for?

File: cmd/extensions/cmd.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/aerogear/charmil/cmd/extensions/installed"
+	"github.com/aerogear/charmil/cmd/extensions/list"
+	"github.com/aerogear/charmil/cmd/extensions/remove"
+	"github.com/spf13/cobra"
+)
+
+// ExtensionsCmd represents the extensions command
+var ExtensionsCmd = &cobra.Command{
+	Use:   "extensions",
+	Short: "A brief description of your command",
```
Comment (wtrocki) [inline cmd/extensions/cmd.go:16]: Missing descriptions

File: cmd/extensions/cmd.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("extensions called")
+	},
+}
+
+func init() {
+	ExtensionsCmd.AddCommand(install.InstallCmd, installed.InstalledCmd, list.ListCmd, remove.RemoveCmd)
+	// Here you will define your flags and configuration settings.
+
+	// Cobra supports Persistent Flags which will work for this command
```
Comment (wtrocki) [inline cmd/extensions/cmd.go:32]: Remove comments

File: cmd/extensions/installed/cmd.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// InstalledCmd represents the installed command
+var InstalledCmd = &cobra.Command{
+	Use:   "installed",
```
Comment (wtrocki) [inline cmd/extensions/installed/cmd.go:11]: That is why I was recommending to check kube lib.
Installed might be simple list and remote would be search comamnd

**Comment by wtrocki** [comment]
> Adding flags to some commands for accepting input. For example: Accepting the name and version values through the command - yourCLi extensions install <name>@<version>
Cleaning up the code by removing some unwanted init() functions and flags
Giving proper descriptions to each command and even adding it to the documentation

Why not making good PR in first place. This is not completion. Good code and overal plan matters more than pushing some generated commands. 
I will merge this PR as exception but let's chat about overal plan rather than add some random changes to repo

**Comment by wtrocki** [comment]
Btw.. this PR introduced wrong structure.

Extensions should be in pkg. please learn basic of golang

**Comment by namit-chandwani** [comment]
> > Adding flags to some commands for accepting input. For example: Accepting the name and version values through the command - yourCLi extensions install @
> > Cleaning up the code by removing some unwanted init() functions and flags
> > Giving proper descriptions to each command and even adding it to the documentation
> 
> Why not making good PR in first place. This is not completion. Good code and overal plan matters more than pushing some generated commands.
> I will merge this PR as exception but let's chat about overal plan rather than add some random changes to repo

Sorry, I will keep this in mind and make sure that I don't repeat this mistake in any of my future PRs. Will create a PR only after I'm 100% confident that it's complete.

I'm making the required changes now, will create a new PR soon.

**Pull request merged**